app/controllers/cinema_halls_controller.rb

```diff
@@ -36,7 +37,7 @@ def update
   end
 
   def destroy
-    @cinema_hall = CinemaHall.find(params[:id])
+    @cinema_hall = CinemaHalls::UseCases::Delete.new(id: params[:id]).call
     @cinema_hall.destroy
```

Comment:
The previous line should delete cinema_hall as I understand.

Reply:
Exactly. Deletion logic should be moved to UseCase class.